README.md
# rosary project

It's implementation open source Progressive
Web App (PWA) that helps to connect people on holy rosary prayer. You
can add intention and Orare Pro Me community will pray on Holy Rosary for you.

Production App consumes backend service, that allows to connect prayers together
(detail's are described in the app).

In development use json-server to set up local API backend

```
npm install -g json-server
```

Once json-server is installed globally, you can run api

```
npm run json-server
```

make sure `.env.development.local` point's to json-server

```
REACT_APP_ROSARY_API_="https://localhost:3001/"
```


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [0.6.0](https://github.com/gom3s/rosary/compare/v0.3.0...v0.6.0) (2020-11-27)


### Features

* **#14:** added repository helper ([461274b](https://github.com/gom3s/rosary/commit/461274bab01d7f2bede3f676282f70d06c1879b3)), closes [#14](https://github.com/gom3s/rosary/issues/14)
* **#14:** added setter to repository helper ([81210cd](https://github.com/gom3s/rosary/commit/81210cdaf933150080dc45d8f34cf622b9bd5e75)), closes [#14](https://github.com/gom3s/rosary/issues/14)
* **#14:** basic intention statistics ([7c6ca36](https://github.com/gom3s/rosary/commit/7c6ca361fa05cb4393a5657120246e1595208f11)), closes [#14](https://github.com/gom3s/rosary/issues/14)
* **#14:** intention statistic ([19c36c1](https://github.com/gom3s/rosary/commit/19c36c1f8258a80e7188ddda0b8f5f785818e0c4)), closes [#14](https://github.com/gom3s/rosary/issues/14)
* **#14:** use local storage for active prayer data ([591b724](https://github.com/gom3s/rosary/commit/591b7241521b440009d89000133690c8f608a8e0)), closes [#14](https://github.com/gom3s/rosary/issues/14)
* **#16:** handle API errors in login register card ([9e6255e](https://github.com/gom3s/rosary/commit/9e6255e13efc7f94eee7da5f6b608f694ead7bc3)), closes [#16](https://github.com/gom3s/rosary/issues/16)
* **#16:** validation added to forms ([cfbe608](https://github.com/gom3s/rosary/commit/cfbe6080b5b3c48927d1caa8b8b41bf9a6d5b838)), closes [#16](https://github.com/gom3s/rosary/issues/16)
* **#23:** add Register card ([77588e6](https://github.com/gom3s/rosary/commit/77588e63f6875cf95115f393d59573248e960481)), closes [#23](https://github.com/gom3s/rosary/issues/23)
* **#23:** added progress bar to register form ([5fd8542](https://github.com/gom3s/rosary/commit/5fd854223e97754bd74c8efc6332de18a020222e)), closes [#23](https://github.com/gom3s/rosary/issues/23)
* **#23:** display error message on login card ([e8a478d](https://github.com/gom3s/rosary/commit/e8a478d41aefddd9411006e6824bcac98b28e4c6)), closes [#23](https://github.com/gom3s/rosary/issues/23)
* **#23:** register form error handling ([11c65fd](https://github.com/gom3s/rosary/commit/11c65fd68f14b0a5862e9ebd91568570653f5dcf)), closes [#23](https://github.com/gom3s/rosary/issues/23)
* **#32:** added delete intention dialog ([6f3954d](https://github.com/gom3s/rosary/commit/6f3954d41d88c5d126e26b590d2911159b65869e)), closes [#32](https://github.com/gom3s/rosary/issues/32)
* **#32:** As Admin I can delete Intention ([454fcf0](https://github.com/gom3s/rosary/commit/454fcf0a4f2236c2e8eda80bcdefe6cdce693c78)), closes [#32](https://github.com/gom3s/rosary/issues/32)
* **#32:** display delete button ([2bf8667](https://github.com/gom3s/rosary/commit/2bf866734e8a457bee61bd0e340d37b1a75cc8fb)), closes [#32](https://github.com/gom3s/rosary/issues/32)
* **#32:** handle delete intention dialog display ([7660108](https://github.com/gom3s/rosary/commit/76601087249ec62582244319ee577fbfbe6ba7ca)), closes [#32](https://github.com/gom3s/rosary/issues/32)
* **#32:** pass intentionId to onDelete action ([62b375e](https://github.com/gom3s/rosary/commit/62b375e118a731d50a4fffbf5d3c288c6cbc5b9f)), closes [#32](https://github.com/gom3s/rosary/issues/32)
* **#37:** Add "how it works" on homepage ([c0a5549](https://github.com/gom3s/rosary/commit/c0a5549b8f8a7715a12128c363dd43071a473e75)), closes [#37](https://github.com/gom3s/rosary/issues/37)
* **#37:** dummy component and route setup ([02aada8](https://github.com/gom3s/rosary/commit/02aada82861fd90be1100aed4af4caf7e9593b02)), closes [#37](https://github.com/gom3s/rosary/issues/37)
* **#38:** redirect after intention save ([ed48688](https://github.com/gom3s/rosary/commit/ed486886bdcfe92c20f5426e6a54453a44caf48d)), closes [#38](https://github.com/gom3s/rosary/issues/38)
* **#38:** secure delete intention with JWT ([59b67ff](https://github.com/gom3s/rosary/commit/59b67ff8e8f4c84c07d7b90fde56b668fe3a0b9d)), closes [#38](https://github.com/gom3s/rosary/issues/38)
* **#58:** added google sign in button ([392a29a](https://github.com/gom3s/rosary/commit/392a29a22415554568f22cb09ef8538ae73d4cd4)), closes [#58](https://github.com/gom3s/rosary/issues/58)
* **#58:** redirect after  login with google ([1242595](https://github.com/gom3s/rosary/commit/1242595664e3d5e235255727119d681207fc46f9)), closes [#58](https://github.com/gom3s/rosary/issues/58)
* **#7:** active prayer saved in context ([d811264](https://github.com/gom3s/rosary/commit/d811264042f93638fa44636720b6b59778f6c6f1)), closes [#7](https://github.com/gom3s/rosary/issues/7)
* **#7:** added progress with label component ([b0305e4](https://github.com/gom3s/rosary/commit/b0305e4f729f433fa9193fb51643e56b6f9447de)), closes [#7](https://github.com/gom3s/rosary/issues/7)
* **#7:** changed suspense indicator on pray req ([1a354a6](https://github.com/gom3s/rosary/commit/1a354a678e3dd9b5ac3b8815d145e1b55995216a)), closes [#7](https://github.com/gom3s/rosary/issues/7)
* **#7:** default active prayer ([fc63cfa](https://github.com/gom3s/rosary/commit/fc63cfaee4f183b1c7d8b5ca009ce76638362ec0)), closes [#7](https://github.com/gom3s/rosary/issues/7)
* **#7:** get active prayer from context ([f6617bb](https://github.com/gom3s/rosary/commit/f6617bbbdc13639953590908e88805fc6d465989)), closes [#7](https://github.com/gom3s/rosary/issues/7)
* **#7:** pray time left progress bar ([7972fa7](https://github.com/gom3s/rosary/commit/7972fa74a308bd794864c862256209f62362c349)), closes [#7](https://github.com/gom3s/rosary/issues/7)
* **#7:** save active prayer in context ([4a42ada](https://github.com/gom3s/rosary/commit/4a42adabddc653af115bc516ed041ea656ee1fee)), closes [#7](https://github.com/gom3s/rosary/issues/7)
* **authProvider:** checks if user has a role ([4133983](https://github.com/gom3s/rosary/commit/41339836dbf8456fac4ca7898ddc3491b41ada7d))

src/consts/rosary_pl.js
import {MysteryTypes} from './MysteryTypes'

/* tslint:disable:object-literal-sort-keys */
export const rosary = [
  {
    type: MysteryTypes.Joyful1,
    image: '',
    title: '1 tajemnica radosna: Zwiastowanie Najświętszej Maryi Pannie',
    description:
      'Łk 1,26-38: W szóstym miesiącu posłał Bóg anioła Gabriela do miasta w Galilei, zwanego Nazaret, do Dziewicy poślubionej mężowi, imieniem Józef, z rodu Dawida; a Dziewicy było na imię Maryja. Anioł wszedł do Niej i rzekł: "Bądź pozdrowiona, pełna łaski, Pan z Tobą, błogosławiona jesteś między niewiastami".',
  },
  {
    type: MysteryTypes.Joyful2,
    image: '',
    title:
      '2 tajemnica radosna: Nawiedzenie św. Elżbiety przez Najświętszą Maryję Pannę',
    description:
      'Łk 1,39-45: W tym czasie Maryja wybrała się i poszła z pośpiechem w góry do pewnego miasta w pokoleniu Judy. Weszła do domu Zachariasza i pozdrowiła Elżbietę...',
  },
  {
    type: MysteryTypes.Joyful3,
    image: '',
    title: '3 tajemnica radosna: Narodzenie Pana Jezusa',
    description:
      ' Łk 2,1-7: W owym czasie wyszło rozporządzenie Cezara Augusta, żeby przeprowadzić spis ludności w całym państwie. Pierwszy ten spis odbył się wówczas, gdy wielkorządcą Syrii był Kwiryniusz...',
  },
  {
    type: MysteryTypes.Joyful4,
    image: '',
    title: '4 tajemnica radosna: Ofiarowanie Pana Jezusa w Świątyni',
    description:
      ' Łk 2,22-38: Gdy potem upłynęły dni ich oczyszczenia według Prawa Mojżeszowego, przynieśli Je do Jerozolimy, aby Je przedstawić Panu. Tak bowiem jest napisane w Prawie Pańskim: Każde pierworodne dziecko płci męskiej będzie poświęcone Panu...',
  },
  {
    type: MysteryTypes.Joyful5,
    image: '',
    title: '5 tajemnica radosna: Znalezienie Pana Jezusa w Świątyni',
    description:
      'Łk 2,41-52: Rodzice Jego chodzili co roku do Jerozolimy na Święto Paschy. Gdy miał lat dwanaście, udali się tam zwyczajem świątecznym. Kiedy wracali po skończonych uroczystościach, został Jezus w Jerozolimie, a tego nie zauważyli Jego Rodzice. Przypuszczając, że jest w towarzystwie pątników, uszli dzień drogi i szukali Go wśród krewnych i znajomych. Gdy Go nie znaleźli, wrócili do Jerozolimy szukając Go.',
  },
  {
    type: MysteryTypes.Luminous1,
    image: '',
    title: '1 tajemnica światła: Chrzest Pana Jezusa w Jordanie',
    description:
      'Łk 3,15-22: Gdy więc lud oczekiwał z napięciem i wszyscy snuli domysły w sercach co do Jana, czy nie jest Mesjaszem, on tak przemówił do wszystkich: "Ja was chrzczę wodą; lecz idzie mocniejszy ode mnie, któremu nie jestem godzien rozwiązać rzemyka u sandałów. On chrzcić was będzie Duchem Świętym i ogniem.',
  },
  {
    type: MysteryTypes.Luminous2,
    image: '',
    title: '2 tajemnica światła: Objawienie się Pana Jezusa na weselu w Kanie',
    description:
      'J 2,1-11: Trzeciego dnia odbywało się wesele w Kanie Galilejskiej i była tam Matka Jezusa. Zaproszono na to wesele także Jezusa i Jego uczniów. A kiedy zabrakło wina, Matka Jezusa mówi do Niego: "Nie mają już wina".',
  },
  {
    type: MysteryTypes.Luminous3,
    image: '',
    title:
      '3 tajemnica światła: Głoszenie Królestwa Bożego i wzywanie do nawrócenia',
    description:
      'Mt 4,23-25: I obchodził Jezus całą Galileę, nauczając w tamtejszych synagogach, głosząc Ewangelię o królestwie i lecząc wszystkie choroby i wszelkie słabości wśród ludu. A wieść o Nim rozeszła się po całej Syrii. Przynoszono więc do Niego wszystkich cierpiących, których dręczyły rozmaite choroby i dolegliwości, opętanych, epileptyków i paralityków, a On ich uzdrawiał. I szły za Nim liczne tłumy z Galilei i z Dekapolu, z Jerozolimy, z Judei i z Zajordania.',
  },
  {
    type: MysteryTypes.Luminous4,
    image: '',
    title: '4 tajemnica światła: Przemienienie się Pana Jezusa na Górze Tabor',
    description:
      'Mt 17,1-8: Po sześciu dniach Jezus wziął z sobą Piotra, Jakuba i brata jego Jana i zaprowadził ich na górę wysoką, osobno. Tam przemienił się wobec nich: twarz Jego zajaśniała jak słońce, odzienie zaś stało się białe jak światło. A oto im się ukazali Mojżesz i Eliasz, którzy rozmawiali z Nim. Wtedy Piotr rzekł do Jezusa: "Panie, dobrze, że tu jesteśmy; jeśli chcesz, postawię tu trzy namioty: jeden dla Ciebie, jeden dla Mojżesza i jeden dla Eliasza". Gdy on jeszcze mówił, oto obłok świetlany osłonił ich, a z obłoku odezwał się głos: "To jest mój Syn umiłowany, w którym mam upodobanie, Jego słuchajcie!" Uczniowie, słysząc to, upadli na twarz i bardzo się zlękli. A Jezus zbliżył się do nich, dotknął ich i rzekł: "Wstańcie, nie lękajcie się!" Gdy podnieśli oczy, nikogo nie widzieli, tylko samego Jezusa.',
  },
  {
    type: MysteryTypes.Luminous5,
    image: '',
    title:
      '5 tajemnica światła: Ustanowienie Eucharystii będącej sakramentalnym wyrazem misterium paschalnego',
    description:
      'Mk 14,22-25: A gdy jedli, wziął chleb, odmówił błogosławieństwo, połamał i dał im mówiąc: "Bierzcie, to jest Ciało moje". Potem wziął kielich i odmówiwszy dziękczynienie dał im, i pili z niego wszyscy. I rzekł do nich: "To jest moja Krew Przymierza, która za wielu będzie wylana. Zaprawdę, powiadam wam: Odtąd nie będę już pił z owocu winnego krzewu aż do owego dnia, kiedy pić go będę nowy w królestwie Bożym".',
  },
  {
    type: MysteryTypes.Sorrowful1,
    image: '',
    title: '1 tajemnica bolesna: Modlitwa Pana Jezusa w Ogrójcu',
    description:
      'Mk 14,32-42: A kiedy przyszli do ogrodu zwanego Getsemani, rzekł Jezus do swoich uczniów: "Usiądźcie tutaj, Ja tymczasem będę się modlił". Wziął ze sobą Piotra, Jakuba i Jana i począł drżeć, i odczuwać trwogę. "I rzekł do nich: "Smutna jest moja dusza aż do śmierci; zostańcie tu i czuwajcie!"',
  },
  {
    type: MysteryTypes.Sorrowful2,
    image: '',
    title: '2 tajemnica bolesna: Biczowanie Pana Jezusa',
    description:
      'Mt 27,15-26: A był zwyczaj, że na każde święto namiestnik uwalniał jednego więźnia, którego chcieli. Trzymano zaś wtedy znacznego więźnia, imieniem Barabasz. Gdy się więc zebrali, spytał ich Piłat: "Którego chcecie, żebym wam uwolnił, Barabasza czy Jezusa, zwanego Mesjaszem?"',
  },
  {
    type: MysteryTypes.Sorrowful3,
    image: '',
    title: '3 tajemnica bolesna: Cierniem ukoronowanie Pana Jezusa',
    description:
      'Mk 15,16-20: Żołnierze zaprowadzili Go na wewnętrzny dziedziniec, czyli pretorium, i zwołali całą kohortę. Ubrali Go w purpurę i uplótłszy wieniec z ciernia włożyli Mu na głowę. I zaczęli Go pozdrawiać: "Witaj, Królu Żydowski!" Przy tym bili Go trzciną po głowie, pluli na Niego i przyklękając oddawali Mu hołd. A gdy Go wyszydzili, zdjęli z Niego purpurę i włożyli na Niego własne Jego szaty.',
  },
  {
    type: MysteryTypes.Sorrowful4,
    image: '',
    title: '4 tajemnica bolesna: Droga Krzyżowa Pana Jezusa',
    description:
      ' Łk 23,26-32: Gdy Go wyprowadzili, zatrzymali niejakiego Szymona z Cyreny, który wracał z pola, i włożyli na niego krzyż, aby go niósł za Jezusem. A szło za Nim mnóstwo ludu, także kobiet, które zawodziły i płakały nad Nim. Lecz Jezus zwrócił się do nich i rzekł: "Córki jerozolimskie, nie płaczcie nade Mną; płaczcie raczej nad sobą i nad waszymi dziećmi! Oto bowiem przyjdą dni, kiedy mówić będą: "Szczęśliwe niepłodne łona, które nie rodziły, i piersi, które nie karmiły". Wtedy zaczną wołać do gór: Padnijcie na nas; a do pagórków: Przykryjcie nas! Bo jeśli z zielonym drzewem to czynią, cóż się stanie z suchym?" Przyprowadzono też dwóch innych - złoczyńców, aby ich z Nim stracić.',
  },
  {
    type: MysteryTypes.Sorrowful5,
    image: '',
    title: '5 tajemnica bolesna: Ukrzyżowanie Pana Jezusa',
    description:
      'J 19,25-30: A obok krzyża Jezusowego stały: Matka Jego i siostra Matki Jego, Maria, żona Kleofasa, i Maria Magdalena. Kiedy więc Jezus ujrzał Matkę i stojącego obok Niej ucznia, którego miłował, rzekł do Matki: "Niewiasto, oto syn Twój". Następnie rzekł do ucznia: "Oto Matka twoja". I od tej godziny uczeń wziął Ją do siebie. Potem Jezus świadom, że już wszystko się dokonało, aby się wypełniło Pismo, rzekł: "Pragnę". Stało tam naczynie pełne octu. Nałożono więc na hizop gąbkę pełną octu i do ust Mu podano. A gdy Jezus skosztował octu, rzekł: "Wykonało się!" I skłoniwszy głowę oddał ducha.',
  },
  {
    type: MysteryTypes.Glorius1,
    image: '',
    title: '1 tajemnica chwalebna: Zmartwychwstanie Pana Jezusa',
    description:
      'J 20,1-18: A pierwszego dnia po szabacie, wczesnym rankiem, gdy jeszcze było ciemno, Maria Magdalena udała się do grobu i zobaczyła kamień odsunięty od grobu. Pobiegła więc i przybyła do Szymona Piotra i do drugiego ucznia, którego Jezus kochał, i rzekła do nich: "Zabrano Pana z grobu i nie wiemy, gdzie Go położono".',
  },
  {
    type: MysteryTypes.Glorius2,
    image: '',
    title: '2 tajemnica chwalebna: Wniebowstąpienie Pana Jezusa',
    description:
      'Łk 24,44-53: Potem rzekł do nich: "To właśnie znaczyły słowa, które mówiłem do was, gdy byłem jeszcze z wami: Musi się wypełnić wszystko, co napisane jest o Mnie w Prawie Mojżesza, u Proroków i w Psalmach".',
  },
  {
    type: MysteryTypes.Glorius3,
    image: '',
    title: '3 tajemnica chwalebna: Zesłanie Ducha Świętego',
    description:
      'Dz 2,1-13: Kiedy nadszedł wreszcie dzień Pięćdziesiątnicy, znajdowali się wszyscy razem na tym samym miejscu. Nagle dał się słyszeć z nieba szum, jakby uderzenie gwałtownego wiatru, i napełnił cały dom, w którym przebywali. Ukazały się im też języki jakby z ognia, które się rozdzieliły, i na każdym z nich spoczął jeden. I wszyscy zostali napełnieni Duchem Świętym, i zaczęli mówić obcymi językami, tak jak im Duch pozwalał mówić.',
  },
  {
    type: MysteryTypes.Glorius4,
    image: '',
    title: '4 tajemnica chwalebna: Wniebowzięcie Najświętszej Maryi Panny',
    description:
      'k 1,46-56: Wtedy Maryja rzekła: Wielbi dusza moja Pana, i raduje się duch mój w Bogu, moim Zbawcy. Bo wejrzał na uniżenie Służebnicy swojej. Oto bowiem błogosławić mnie będą odtąd wszystkie pokolenia, gdyż wielkie rzeczy uczynił mi Wszechmocny. Święte jest Jego imię - a swoje miłosierdzie na pokolenia i pokolenia zachowuje dla tych, co się Go boją. ',
  },
  {
    type: MysteryTypes.Glorius5,
    image: '',
    title:
      '5 tajemnica chwalebna: Ukoronowanie Najświętszej Maryi Panny na Królową Nieba i Ziemi',
    description:
      'Ap 12,1: Potem wielki znak się ukazał na niebie: Niewiasta obleczona w słońce i księżyc pod jej stopami a na jej głowie wieniec z gwiazd dwunastu.',
  },
]
